buildSrc/src/main/java/org/elasticsearch/gradle/LazyString.java

```diff
@@ -17,15 +17,25 @@
  * under the License.
  */
 
-package org.elasticsearch.packaging.test;
+package org.elasticsearch.gradle;
 
-import org.elasticsearch.packaging.util.Distribution;
+import java.util.function.Supplier;
 
-public class DefaultNoJdkDebBasicTests extends PackageTestCase {
+/**
+ * An functional interface to allow a String supplier to be resolved when calling toString().
```

Comment:
nit: this isn't really a functional interface in the pedantic sense of `@FunctionalInterface`.